### PR TITLE
Improvement/added flow

### PR DESCRIPTION
**Pull request by rfreitas**
# Problem

There was no type checking.
Also the library didn't work with react, only preact.

# Solution

Added flow (upgraded webpack and babel in the process).
Fixed the example html, it is now using preact compat

### CODE REVIEW

File: src/react-webcam.js [hunk 1]
```diff
@@ -152,34 +162,29 @@ export default class Webcam extends Component {
     Webcam.mountedInstances.splice(index, 1);
 
     if (Webcam.mountedInstances.length === 0 && this.state.hasUserMedia) {
-      if (this.stream.stop) {
-        this.stream.stop();
```
Comment (rfreitas) [inline src/react-webcam.js]: `stop` has been deprecated in MediaStream. Flow told me that!

File: package.json [hunk 2]
```diff
@@ -5,9 +5,11 @@
   "main": "dist/react-webcam.js",
   "scripts": {
     "prepublish": "npm run build:umd",
```
Comment (jenniferjentle) [inline package.json]: Where is this `prepublish` script called? It's calling `build:umd` but you have deleted it.

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  # replaces non alphanumeric symbols with "-"
+  # sed -r is only supported in linux, ref http://stackoverflow.com/a/2871217/689223
+  # Domain names follow the RFC1123 spec [a-Z] [0-9] [-]
+  # The length is limited to 253 characters
+  # https://en.wikipedia.org/wiki/Domain_Name_System#Domain_name_syntax
+  DEPLOY_SUBDOMAIN=`echo "$DEPLOY_SUBDOMAIN_UNFORMATTED" | sed -r 's/[^A-Za-z0-9]+/\-/g'`
+  echo $DEPLOY_SUBDOMAIN
+
+  DEPLOY_DOMAIN=https://${DEPLOY_SUBDOMAIN}-${REPO_NAME}-${REPO_OWNER}.surge.sh
```
Comment (jenniferjentle) [inline deploy.sh]: Will this allow to test the examples on a surge link?

File: src/react-webcam.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
-import React, { Component, PropTypes } from 'react';
+/* @flow */
```
Comment (jenniferjentle) [inline src/react-webcam.js]: Is this a magic comment? What does it do?

File: deploy.sh (hunk 3 above)
Comment (rfreitas) [reply deploy.sh]: yes, but I need to add the credentials to Travis. And first, I need to find them :P

File: package.json (hunk 2 above)
Comment (rfreitas) [reply package.json]: thanks for the catch!

File: src/react-webcam.js (hunk 4 above)
Comment (rfreitas) [reply src/react-webcam.js]: yes. turns on flow for the file. It's sad it's like that..

File: src/react-webcam.js [hunk 5]
```diff
@@ -7,27 +8,31 @@ const mediaDevices = navigator.mediaDevices;
...
-      'image/jpeg'
-    ]),
-    className: PropTypes.string,
-    audioSource: PropTypes.string,
-    videoSource: PropTypes.string
-  };
+type constraintTypes = number | Object;
+
+type CameraType = {
+  audio?: boolean,
```
[the remark this answers is not in the document]
Comment (rfreitas) [reply src/react-webcam.js]: it means it's an optional prop

**Pull request merged**